dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -1182,13 +1182,13 @@
 	<Target Name="_ComputeDynamicLibrariesToReidentify">
 		<PropertyGroup>
 			<_ExecutablePathPrefix Condition="'$(_PlatformName)' == 'iOS' Or '$(_PlatformName)' == 'tvOS'">@executable_path/</_ExecutablePathPrefix>
-			<_ExecutablePathPrefix Condition="'$(_PlatformName)' == 'macOS' Or '$(_PlatformName)' == 'MacCatalyst'">@executable_path/../$(_CustomBundleName)/</_ExecutablePathPrefix>
+			<_ExecutablePathPrefix Condition="'$(_PlatformName)' == 'macOS' Or '$(_PlatformName)' == 'MacCatalyst'">@executable_path/../../</_ExecutablePathPrefix>
 		</PropertyGroup>
 		<ItemGroup>
 			<_DynamicLibraryToReidentify Include="@(_FileNativeReference)" Condition="'%(Extension)' == '.dylib'">
-				<DynamicLibraryId>$(_ExecutablePathPrefix)%(Filename)%(Extension)</DynamicLibraryId>
 				<SourceItemGroup>_FileNativeReference</SourceItemGroup>
-				<ReidentifiedPath>$(_IntermediateNativeLibraryDir)/%(Filename)%(Extension)</ReidentifiedPath>
+				<DynamicLibraryId Condition="'%(_FileNativeReference.DynamicLibraryId)' == ''">$([System.String]::Copy('$(_ExecutablePathPrefix)%(_FileNativeReference.RelativePath)').Replace('\', '/'))</DynamicLibraryId>
+				<ReidentifiedPath Condition="'%(_FileNativeReference.ReidentifiedPath)' == ''">$(_IntermediateNativeLibraryDir)/%(_FileNativeReference.RelativePath)</ReidentifiedPath>
```

Comment:
One of our tests fail, because RelativePath isn't always set:

```
../../../test-libraries/.libs/ios-fat/libtest.dylib
    SourceItemGroup = _FileNativeReference
    ReidentifiedPath = obj/Debug/net7.0-ios/iossimulator-x64/nativelibraries//
    Kind = Dynamic
    DynamicLibraryId = @executable_path/
```

You can see the full task output here: https://gist.github.com/rolfbjarne/22209207b141b395aba164c181ff6737

Reply:
Given that macOS and iOS need to be treated differently (the iOS case does not need to take in account relative paths) I have split the logic in 2 where the iOS case is unchanged. Let me know what you think.

Reply:
It doesn't work, because we have a macOS version of the same test that fails for the same reason: RelativePath isn't set.

Maybe computing a value from `%(Filename)%(Extension)` if `%(RelativePath)` isn't set could work:

```xml
<!-- Create the ComputedRelativePath metadata from RelativePath -->
<_DynamicLibraryToReidentify Include="@(_FileNativeReference)" Condition="'%(Extension)' == '.dylib'">
	<SourceItemGroup>_FileNativeReference</SourceItemGroup>
	<ComputedRelativePath>$(_FileNativeReference.RelativePath)</ComputedRelativePath>
</_DynamicLibraryToReidentify>
<!-- Set ComputedRelativePath from Filename and Extension if not set (i.e. RelativePath didn't exist on the item) -->
<_DynamicLibraryToReidentify Update="@(_DynamicLibraryToReidentify)" Condition="'%(SourceItemGroup)' == '_FileNativeReference' And '%(_DynamicLibraryToReidentify.ComputedRelativePath)' == ''">
	<ComputedRelativePath>%(Filename)%(Extension)</ComputedRelativePath>
</_DynamicLibraryToReidentify>
<!-- Compute DynamicLibraryId and ReidentifiedPath -->
<_DynamicLibraryToReidentify Update="@(_DynamicLibraryToReidentify)" Condition="'%(SourceItemGroup)' == '_FileNativeReference'">
	<DynamicLibraryId Condition="'%(_DynamicLibraryToReidentify.DynamicLibraryId)' == ''">$([System.String]::Copy('$(_ExecutablePathPrefix)%(_DynamicLibraryToReidentify.ComputedRelativePath)').Replace('\', '/'))</DynamicLibraryId>
	<ReidentifiedPath Condition="'%(_DynamicLibraryToReidentify.ReidentifiedPath)' == ''">$(_IntermediateNativeLibraryDir)/%(_DynamicLibraryToReidentify.ComputedRelativePath)</ReidentifiedPath>
</_DynamicLibraryToReidentify>
```